src/http_server.cc

```diff
@@ -4105,8 +4173,17 @@ HTTPAPIServer::GenerateRequestClass::EndResponseCallback(
   auto infer_request =
       reinterpret_cast<HTTPAPIServer::GenerateRequestClass*>(arg);
 
-  infer_request->SendChunkResponse(true /* end */);
-  evhtp_send_reply_chunk_end(infer_request->EvHtpRequest());
+  if (infer_request->req_ == nullptr) {
+    if (infer_request->triton_request_ != nullptr) {
+      LOG_TRITONSERVER_ERROR(
+          TRITONSERVER_InferenceRequestCancel(infer_request->triton_request_),
+          "cancelling request");
+    }
+  } else {
+    infer_request->SendChunkResponse(true /* end */);
+    evhtp_send_reply_chunk_end(infer_request->EvHtpRequest());
```

Comment:
Not a blocking comment for this PR - but could be a non-cherry-picked follow up. 

Are `infer_reqest->req_` and `infer_request->EvHtpRequest()` the same object? Seems like they're used interchangeably or sometimes both used in the same function like this one.

Reply:
they are the same. previously the callbacks were accessing the object from a different class and so `infer_request->EvHtpRequest()` was necessary but with the refactor to move the callbacks to the infer class - I think we can remove it. will handle in follow up PR.